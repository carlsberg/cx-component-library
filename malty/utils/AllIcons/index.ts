export { allIcons, allIconsStoryOptions } from './AllIconsMap';
