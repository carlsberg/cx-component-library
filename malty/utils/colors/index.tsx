const rgbToHex = (value: string): string =>
  value.replace(/rgb\((\d+),\s*(\d+),\s*(\d+)\)/gi, (_, r, g, b) => {
    const toHex = (n: string) => parseInt(n, 10).toString(16).padStart(2, '0');
    return `#${toHex(r)}${toHex(g)}${toHex(b)}`;
  });

export { rgbToHex };
