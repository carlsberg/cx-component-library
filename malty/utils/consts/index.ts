export enum EventKeys {
  ENTER = 'Enter',
  SPACE = 'Space',
  ESCAPE = 'Escape'
}
