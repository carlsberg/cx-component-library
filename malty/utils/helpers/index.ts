export const getBreakpointNumber = (breakpoint: string) => Number(breakpoint.split('px')[0]);
