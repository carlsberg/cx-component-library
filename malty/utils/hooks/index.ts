export { Device, useMatchMedia } from './useMatchMedia';
