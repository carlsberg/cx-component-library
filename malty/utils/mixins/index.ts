import { css } from 'styled-components';

// Hide an element to all devices except screen readers. More info here: https://webaim.org/techniques/css/invisiblecontent/
export const srOnly = css`
  position: absolute;
  left: -10000px;
  top: auto;
  width: 1px;
  height: 1px;
  overflow: hidden;
`;
