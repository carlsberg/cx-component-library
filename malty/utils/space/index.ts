import type { ArgTypes } from '@storybook/react';

export interface SpaceProps {
  m?: string;
  mt?: string;
  mr?: string;
  mb?: string;
  ml?: string;
  mx?: string;
  my?: string;
}

export const space = <Props extends SpaceProps>({ m = '', mx, my, mt, mr, mb, ml }: Props) => `
    margin: ${m};
    margin-top: ${mt ?? my ?? ''};
    margin-bottom: ${mb ?? my ?? ''};
    margin-right: ${mr ?? mx ?? ''};
    margin-left: ${ml ?? mx ?? ''};
`;

export function spreadSpaceProps<Props extends SpaceProps>(props: Props) {
  const { m, mt, mr, mb, ml, mx, my, ...restProps } = props;

  return { spaceProps: { m, mt, mr, mb, ml, mx, my }, restProps };
}

export function generateStorybookSpacing(): ArgTypes {
  return {
    m: {
      description: 'margin',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    },
    mt: {
      description: 'margin-top',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    },
    mr: {
      description: 'margin-right',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    },
    mb: {
      description: 'margin-bottom',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    },
    ml: {
      description: 'margin-left',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    },
    mx: {
      description: 'margin-left, margin-right',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    },
    my: {
      description: 'margin-top, margin-bottom',
      control: 'text',
      table: {
        category: 'Spacing'
      }
    }
  };
}
