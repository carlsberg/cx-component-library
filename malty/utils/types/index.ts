export interface SpaceProps {
  m?: string;
  mt?: string;
  mr?: string;
  mb?: string;
  ml?: string;
  mx?: string;
  my?: string;
}
