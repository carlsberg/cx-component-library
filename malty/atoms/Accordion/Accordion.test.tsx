import { fireEvent, jsonRenderer, render, screen } from '@carlsberggroup/malty.utils.test';
import React from 'react';
import { Accordion, AccordionItem } from '.';

jest.mock('uuid', () => ({ v4: () => '00000000-0000-0000-0000-000000000000' }));

describe('Accordion', () => {
  it('matches snapshot', () => {
    const view = jsonRenderer(
      <Accordion>
        <AccordionItem title="Accordion title 1">
          <div>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et
            dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex
            ea commodo consequat.
          </div>
        </AccordionItem>
        <AccordionItem title="Accordion title 2">
          <div>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et
            dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex
            ea commodo consequat.
          </div>
        </AccordionItem>
      </Accordion>
    );
    expect(view).toMatchSnapshot();
  });

  it('renders elements', () => {
    render(
      <Accordion>
        <AccordionItem title="Accordion title 1">
          <div>Accordion content 1</div>
        </AccordionItem>
        <AccordionItem title="Accordion title 2">
          <div>Accordion content 2</div>
        </AccordionItem>
      </Accordion>
    );
    expect(screen.getByText('Accordion title 1')).toBeInTheDocument();
  });
  it('opens accordion on click', () => {
    render(
      <Accordion>
        <AccordionItem title="Accordion title 1">
          <div>Accordion content 1</div>
        </AccordionItem>
        <AccordionItem title="Accordion title 2">
          <div>Accordion content 2</div>
        </AccordionItem>
      </Accordion>
    );
    const accordionItem = screen.getByText('Accordion title 1');
    fireEvent.click(accordionItem);
    const text = screen.queryByText('Accordion content 1');
    expect(text).toBeVisible();
  });
});
