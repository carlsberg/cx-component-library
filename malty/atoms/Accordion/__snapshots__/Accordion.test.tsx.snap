// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion matches snapshot 1`] = `
<span
  className="sc-bdvvtL bWTUHG"
>
  <ul
    className="sc-dkPtRN fQApVx"
    data-testid="undefined-accordion-container"
  >
    <span
      className="sc-bdvvtL bWTUHG"
    >
      <li
        className="sc-hKwDye dgXmE"
        data-testid="undefined-item"
      >
        <div
          aria-controls="accordion-item-\\"00000000-0000-0000-0000-000000000000"
          className="sc-eCImPb kJyoKS accordion-header"
          onClick={[Function]}
          size={32}
        >
          <span
            className="sc-bdvvtL bWTUHG"
          >
            <p
              className="sc-furwcr lnhugg"
              color="digital-black"
            >
              Accordion title 1
            </p>
          </span>
          <svg
            className="sc-gsDKAQ kyUyaQ sc-iCfMLu hBqQrs"
            color="digital-black"
            data-testid="svg-component"
            size="24px"
            version="1.1"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <g
              fillRule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 13.586L6.707 8.293a1 1 0 00-1.414 1.414l6 6a1 1 0 001.414 0l6-6a1 1 0 10-1.414-1.414z"
              />
            </g>
          </svg>
        </div>
        <div
          className="sc-jRQBWg cmRugU accordion-body"
          data-testid="undefined-content-container"
          id="accordion-item-\\"00000000-0000-0000-0000-000000000000"
        >
          <div>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
          </div>
        </div>
      </li>
    </span>
    <span
      className="sc-bdvvtL bWTUHG"
    >
      <li
        className="sc-hKwDye dgXmE"
        data-testid="undefined-item"
      >
        <div
          aria-controls="accordion-item-\\"00000000-0000-0000-0000-000000000000"
          className="sc-eCImPb kJyoKS accordion-header"
          onClick={[Function]}
          size={32}
        >
          <span
            className="sc-bdvvtL bWTUHG"
          >
            <p
              className="sc-furwcr lnhugg"
              color="digital-black"
            >
              Accordion title 2
            </p>
          </span>
          <svg
            className="sc-gsDKAQ kyUyaQ sc-iCfMLu hBqQrs"
            color="digital-black"
            data-testid="svg-component"
            size="24px"
            version="1.1"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <g
              fillRule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 13.586L6.707 8.293a1 1 0 00-1.414 1.414l6 6a1 1 0 001.414 0l6-6a1 1 0 10-1.414-1.414z"
              />
            </g>
          </svg>
        </div>
        <div
          className="sc-jRQBWg cmRugU accordion-body"
          data-testid="undefined-content-container"
          id="accordion-item-\\"00000000-0000-0000-0000-000000000000"
        >
          <div>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
          </div>
        </div>
      </li>
    </span>
  </ul>
</span>
`;
